tachyon/math/finite_fields/packed_prime_field32_neon.h

```diff
@@ -74,21 +83,33 @@ ALWAYS_INLINE uint32x4_t SubMod32(uint32x4_t lhs, uint32x4_t rhs,
   //     diff := t mod 2³²
   //     underflow := 0 if lhs ≥ rhs else 2³² - 1
   //     r := (diff - underflow * p) mod 2³²
+  //     m := { p     if this is montgomery form, which is 0
+  //          { p - 1 otherwise
+  //
+  // 0 ≤ lhs, rhs ≤ m
+  //
+  // 1) lhs = p && rhs ≠ p
+  //    1 ≤ d ≤ p, go to 4)
   //
-  // 0 ≤ lhs, rhs ≤ p - 1
+  // 2) lhs ≠ p && rhs = p
+  //    -p ≤ d ≤ -1, go to 5)
```

Comment:
```suggestion
  //    -p + 1 ≤ d ≤ -1, go to 5)
```

Reply:
I meant that range is -p ≤ d ≤ -1 and -p + 1 ≤ d ≤ -1, which is case 5, includes the range so it has to go to case 5.